test/unit/shipment_packer_test.rb

```diff
@@ -146,13 +146,38 @@ def setup
     assert_equal 100, package.value
   end
 
-  test "excess packages" do
+  test "excess packages raised over threshold before packing begins" do
+    ActiveShipping::Package.expects(:new).never
+    items = [{ grams: 1, quantity: ShipmentPacker::EXCESS_PACKAGE_QUANTITY_THRESHOLD + 1, price: 1.0 }]
+
     assert_raises(ShipmentPacker::ExcessPackageQuantity) do
-      items = [{ grams: 1, quantity: ShipmentPacker::EXCESS_PACKAGE_QUANTITY_THRESHOLD + 1, price: 1.0 }]
       ShipmentPacker.pack(items, @dimensions, 1, 'USD')
     end
   end
 
+  test "excess packages not raised at threshold" do
+    items = [{ grams: 1, quantity: ShipmentPacker::EXCESS_PACKAGE_QUANTITY_THRESHOLD, price: 1.0 }]
+    packages = ShipmentPacker.pack(items, @dimensions, 1, 'USD')
+
+    assert_predicate packages, :present?
+  end
+
+  test "excess packages not raised below threshold" do
+    items = [{ grams: 1, quantity: ShipmentPacker::EXCESS_PACKAGE_QUANTITY_THRESHOLD - 1, price: 1.0 }]
+    packages = ShipmentPacker.pack(items, @dimensions, 1, 'USD')
+
+    assert_predicate packages, :present?
+  end
```

Comment:
maybe 1 more test for grams=1, `quantity: ShipmentPacker::EXCESS_PACKAGE_QUANTITY_THRESHOLD +1`, and `max_weight=1`?

not sure if the other test below covers that (at least it's not as straightforward to read